types/detect/detect.go

```diff
@@ -36,6 +42,16 @@ type SignatureEventSelector struct {
 	Origin string
 }
 
+func (selector SignatureEventSelector) ToFilters() []protocol.Filter {
```

Comment:
So you're basically generating an initial filter from the existing event selector right ? Should we just mandate event selector to be a regular filter and you just get it together with all others ? No hard opinion, thinking out loud.

Reply:
@AsafEitani and I talked about the similarity between the two, which is why he wanted this function to exist, but I think there is merit to separate definitions because they do separate things, the selectors are used to route inside the engine, a functionality still needed even after the filters. But yes, the selectors do allow us to have the most simple filters for all signatures.
I'm actually considering moving this function to `engine.go` to avoid this implied understanding that these two structs are the same.
And defining the selectors as filters would require large code changes in other places, so even if we go that direction, it might be best to think of it later down the line.

Reply:
> I'm actually considering moving this function to engine.go to avoid this implied understanding that these two structs are the same.

I like that as well.